src/main/java/cat/udl/eps/softarch/learnphysics/config/ApplicationContext.java

```diff
@@ -5,58 +5,129 @@
-import java.util.stream.Collectors;
 
 @Component
 public class ApplicationContext implements InitializingBean {
 
     List<Level> levels = new ArrayList<>();
+    //List<Level> topics = new ArrayList<>();
 
     final LevelRepository levelRepository;
 
     final TopicRepository topicRepository;
 
-    public ApplicationContext(LevelRepository levelRepository, TopicRepository topicRepository) {
+    final QuestionRepository questionRepository;
+
+    public ApplicationContext(LevelRepository levelRepository, TopicRepository topicRepository, QuestionRepository questionRepository) {
+
         this.levelRepository = levelRepository;
         this.topicRepository = topicRepository;
+        this.questionRepository = questionRepository;
+    }
+
+    private List<Question> makeQuestions(Integer lvlNum, String name) {
+
+        List<Question> questions = new ArrayList<>();
+        Question question;
+
+        for (int i = 1; i <= 3; i++) {
+
+            String statement = "Q" + Integer.toString(i) + " " + name;
+            System.out.println(statement);
+
+            question = new Question(statement, "yes");
+            //question.setLevelId(t.getLevel());
+            //question = questionRepository.save(question);
+            questions.add(question);
+        }
+
+        questionRepository.saveAll(questions);
+
+        return questions;
     }
 
     private List<Topic> makeTopics(Integer lvlNum, List<String> difficulties) {
+
         List<Topic> topics = new ArrayList<>();
-        List<Question> questions = new ArrayList<>();
+        Topic topic;
+        List<Question> questions;
+
         List<Theory> theory = new ArrayList<>();
-        topics.add(new Topic("Computer Science Theory" + lvlNum, "This topic is " + difficulties.get(lvlNum-1)
-                + "level difficulty to Computer Science Theory", questions, theory));
-        topics.add(new Topic("IT " + lvlNum, "This topic is " + difficulties.get(lvlNum-1)
-                + "level difficulty to IT", questions, theory));
-        topics.add(new Topic("Programming " + lvlNum, "This topic is " + difficulties.get(lvlNum-1)
-                + "level difficulty to Programming", questions, theory));
-        topics.add(new Topic("Hardware " + lvlNum, "This topic is " + difficulties.get(lvlNum-1)
-                + "level difficulty to Hardware", questions, theory));
-        topics.add(new Topic("Build API " + lvlNum, "This topic is " + difficulties.get(lvlNum-1)
-                + "level difficulty to Build API", questions, theory));
+
+        String topic_name;
+
+        topic_name = "Computer Science Theory" + lvlNum;
+        questions = makeQuestions(lvlNum, topic_name);
+        topic = new Topic(topic_name, "This topic is " + difficulties.get(lvlNum-1) + "level difficulty to Computer Science Theory", questions, theory);
+        topics.add(topic);
+
+        for (Question q: questions) {
+            q.setTopic(topic);
+            questionRepository.save(q);
```

Comment:
I know, but roberto told me have to do a push to review an error when application execute content initialize

Reply:
No errors should appear after solving this. I've just pushed a commit with a solution to this and the application starts successfully. If it doesn't, it may be a problem with the database. Removing the `test2` file containing the database should solve it. However, I've also changed the location of the database file so it will create a new file now.